app/backend/src/couchers/servicers/auth.py

```diff
@@ -154,6 +154,12 @@ def SignupFlow(self, request, context):
                     existing_user = session.execute(
                         select(User).where(User.email == request.basic.email)
                     ).scalar_one_or_none()
+                    existing_user_name=session.execute(
+                        select(User).where(User.username == request.basic.username)
+                    ).scalar_one_or_none()
+                    if existing_user_name:
+                        context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.SIGNUP_FLOW_USERNAME_TAKEN)
+
```

Comment:
Move these added lines two lines down so that the email check is not interrupted in the middle.